src/api/services/userService.spec.ts

```diff
@@ -0,0 +1,104 @@
+        const targetParticipant = await createParticipant(knex, {});
+        const anotherParticipant = await createParticipant(knex, {});
+        const user = await createUser({
+          participantToRoles: [
+            { participantId: targetParticipant.id },
+            { participantId: anotherParticipant.id },
+          ],
+        });
+        const request = createUserParticipantRequest(user.email, targetParticipant, user.id);
+        await verifyAndEnrichUser(request, res, next);
+
+        const userService = new UserService();
+        await userService.removeUser(request);
+
+        const userParticipantIds = await getParticipantIdsOfUser(user.email);
+        expect(userParticipantIds).not.toContain(targetParticipant.id);
+        expect(userParticipantIds).toContain(anotherParticipant.id);
+      });
+    });
+    it('user keycloak roles are unchanged', async () => {
+      const targetParticipant = await createParticipant(knex, {});
+      const anotherParticipant = await createParticipant(knex, {});
+      const user = await createUser({
+        participantToRoles: [
+          { participantId: targetParticipant.id },
+          { participantId: anotherParticipant.id },
+        ],
+      });
+      const request = createUserParticipantRequest(user.email, targetParticipant, user.id);
+      await verifyAndEnrichUser(request, res, next);
+
+      const userService = new UserService();
+      await userService.removeUser(request);
+
+      expect(kcUsersService.removeApiParticipantMemberRole).not.toHaveBeenCalled();
+    });
+    describe('User is removed from their only participant', () => {
+      it('removes the user from their only participant', async () => {
+        const participant = await createParticipant(knex, {});
+        const user = await createUser({
+          participantToRoles: [{ participantId: participant.id }],
+        });
+        const request = createUserParticipantRequest(user.email, participant, user.id);
+        await verifyAndEnrichUser(request, res, next);
+
+        const userService = new UserService();
+        await userService.removeUser(request);
+
+        const userParticipantIds = await getParticipantIdsOfUser(user.email);
+        expect(userParticipantIds).not.toContain(participant.id);
+      });
+      it('removes the Keycloak role when the user has no participants left', async () => {
+        const participant = await createParticipant(knex, {});
+        const user = await createUser({
+          participantToRoles: [{ participantId: participant.id }],
+        });
+        const request = createUserParticipantRequest(user.email, participant, user.id);
+        await verifyAndEnrichUser(request, res, next);
+
+        const userService = new UserService();
+        await userService.removeUser(request);
+
+        expect(kcUsersService.removeApiParticipantMemberRole).toHaveBeenCalledWith(
+          expect.anything(),
+          user.email
```

Comment:
Same comment here with reusing code of `participant` and `user`

Reply:
same comment from me too.  :)